### ISSUE TITLE
cannot build ncspot with cargo

### ISSUE THREAD

**Issue by dmaglio** [librespot-org/librespot#460]
```
   Compiling librespot-protocol v0.1.1
error[E0308]: mismatched types
  --> /home/davidemaglio/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:39
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                       ^^^^^^^ expected struct `std::path::Path`, found `str`
   |
   = note: expected reference `&std::path::Path`
              found reference `&'static str`

error[E0308]: mismatched types
  --> /home/davidemaglio/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:51
   |
45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
   |                                                   ^^^ expected struct `std::path::Path`, found struct `std::string::String`
   |
   = note: expected reference `&std::path::Path`
              found reference `&std::string::String`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `librespot-protocol`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `ncspot v0.1.3`, intermediate artifacts can be found at `/tmp/cargo-installbsLCky`

Caused by:
  build failed

```

**Comment by spktkpkt** [comment]
Same here.

	$ cargo install librespot
	...
	   Compiling librespot-protocol v0.1.1
	error[E0308]: mismatched types
	  --> /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:39
	   |
	45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
	   |                                       ^^^^^^^ expected struct `std::path::Path`, found `str`
	   |
	   = note: expected reference `&std::path::Path`
		          found reference `&'static str`

	error[E0308]: mismatched types
	  --> /home/me/.cargo/registry/src/github.com-1ecc6299db9ec823/librespot-protocol-0.1.1/build.rs:45:51
	   |
	45 |         let p = parse_and_typecheck(&["proto"], &[src]).expect("protoc");
	   |                                                   ^^^ expected struct `std::path::Path`, found struct `std::string::String`
	   |
	   = note: expected reference `&std::path::Path`
		          found reference `&std::string::String`

	error: aborting due to 2 previous errors

	For more information about this error, try `rustc --explain E0308`.
	error: could not compile `librespot-protocol`.
	warning: build failed, waiting for other jobs to finish...
	error: failed to compile `librespot v0.1.1`, intermediate artifacts can be found at `/tmp/cargo-installhkxtbM`

	Caused by:
	  build failed

---

	$ rustc --version
	rustc 1.42.0 (b8cedc004 2020-03-09)

---

	$ swupd info 
	Distribution:      Clear Linux OS
	Installed version: 32840

**Comment by otceliker** [comment]
Had the same problem. Cloning the ncspot repository and compiling with `cargo build --release` solved it for me, if it helps.

**Comment by spktkpkt** [comment]
You're right @otceliker, this works. Thank you for the hint.

**Comment by aso824** [comment]
Didn't worked for me - pulled ncpot, after some problems finally compiled with `cargo build --release` but still `cargo install librespot` didn't worked.

**Comment by ashthespy** [comment]
Any other crate that uses `librespot-protocol` will run into this issue, as we don't pin our version, and https://github.com/stepancheg/rust-protobuf/pull/406 breaks our parsing..

**Comment by Johannesd3** [comment]
Seems it was fixed by e97911e3cfcb85b524e844d0ef5417bbee1e4970 8 months ago. We could remove the version restriction again.
